src/nimony/sem.nim

```diff
@@ -7096,6 +7096,61 @@ proc writeOutput(c: var SemContext; outfile: string) =
       toBuild: move c.toBuild,
       exportBuf: buildIndexExports(c))
 
+  # Update .1.deps.nif file that doesn't contain modules imported under `when false:`
+  # so that Hexer and following phases doesn't read such modules.
+  # It is cached and can be read by Nimony to generate frontend Makefile in next build.
+  # So it need to contain include statements.
+  var deps = createTokenBuf(16)
+  deps.buildTree StmtsS, NoLineInfo:
+    if c.meta.includedFiles.len != 0:
+      deps.buildTree IncludeS, NoLineInfo:
+        for i in c.meta.includedFiles:
+          deps.addStrLit i.toAbsolutePath
+    if c.importedModules.len != 0:
+      let systemSymId = if {SkipSystem, IsSystem} * c.moduleFlags == {}:
+                          c.currentScope.tab[pool.strings.getOrIncl("system")][0].name
+                        else:
+                          SymId(0)
+      deps.buildTree ImportS, NoLineInfo:
+        for symId, i in c.importedModules:
+          if symId != systemSymId:
+            deps.addStrLit i.path.toAbsolutePath
+  let depsFile = changeFileExt(infile, ".deps.nif")
```

Comment:
No. Do not update the .deps.nif file. Instead generate an error file.

Reply:
Errors are written to ".idx.nif" in `quitWithError` proc.
When imported module has errors in `.idx.nif`, the errors are written to `.idx.nif` in `importSingleFile` proc in `src/nimony/semimport.nim`.
Updated `.deps.nif` are used to generate final Makefile to avoid Hexer and later phase reading modules imported under `when false:`.
It should write to a separate file (`.2.deps.nif`) instead of updating existing `.1.deps.nif` files?

Reply:
Oh yeah .2.deps.nif makes sense. I think.

Reply:
I changed code to write `.2.deps.nif`.